src/macros.php

```diff
@@ -297,3 +299,49 @@
         return $this->reverse()->after($currentItem, $fallback);
     });
 }
+
+if (! Collection::hasMacro('paginate')) {
+    /*
+     * Paginate the given collection
+     *
+     * @param int $perPage
+     * @param int $total
+     * @param int $page
+     * @param string $pageName
+     *
+     * @return \Illuminate\Pagination\LengthAwarePaginator
+     */
+    Collection::macro('paginate', function (int $perPage = 15, string $pageName = 'page', int $page = null, int $total = null, array $options = []): LengthAwarePaginator {
```

Comment:
Please exactly match the `LengthAwarePaginator`'s signature for the method.

`$items, $total, $perPage, $currentPage = null, array $options = []`